From f799f1bec7456f9c5efed112a854a7b55de8379a Mon Sep 17 00:00:00 2001
From: Tony Ye <tony.ye@intel.com>
Date: Tue, 21 Feb 2017 14:09:24 -0500
Subject: [PATCH 150/153] [VPG]: drm/i915: prevent crash with .disable_display parameter

Dereferencing display.get_display_clock_speed will cause fatal crash
during driver loading with .disable_display parameter.
---
 drivers/gpu/drm/i915/intel_display.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 4b7a768..2f89b64 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5444,6 +5444,9 @@ static void intel_update_cdclk(struct drm_device *dev)
 {
 	struct drm_i915_private *dev_priv = dev->dev_private;
 
+    if (!dev_priv->display.get_display_clock_speed)
+        return;
+
 	dev_priv->cdclk_freq = dev_priv->display.get_display_clock_speed(dev);
 	DRM_DEBUG_DRIVER("Current CD clock rate: %d kHz\n",
 			 dev_priv->cdclk_freq);
-- 
1.7.1

